holidays/countries/hongkong.py

```diff
@@ -198,8 +198,21 @@ def _populate(self, year):
         dt = self.cnls.lunar_to_gre(year, 8, 15)
         mid_autumn_date = date(dt.year, dt.month, dt.day)
         if self.observed:
-            if mid_autumn_date.weekday() == SAT:
+            # if Chinese Mid-Autumn Festival lies on Saturday
+            # before 1983 public holiday lies on Monday
+            # from 1983 to 2010 public holiday lies on same day
+            # since 2011 public holiday lies on Monday
+            if mid_autumn_date.weekday() == SAT and (
+                year <= 2010 and year >= 1983
+            ):
                 self[mid_autumn_date] = name
+            elif mid_autumn_date.weekday() == SAT and (
+                year >= 2011 or year < 1983
+            ):
+                self[mid_autumn_date + rd(days=+2)] = (
+                    "The second day of the " + name + " (Monday)"
+                )
+                mid_autumn_date = mid_autumn_date + rd(days=+2)
```

Comment:
```suggestion
            if mid_autumn_date.weekday() == SAT:
                if 1983 <= year <= 2010:
                    self[mid_autumn_date] = name
                else:
                    self[mid_autumn_date + rd(days=+2)] = (
                        "The second day of the " + name + " (Monday)"
                    )
                    mid_autumn_date = mid_autumn_date + rd(days=+2)
```

Reply:
Thanks for your clear code, updated as below.